xarray/conventions.py

```diff
@@ -756,11 +756,13 @@ def _encode_coordinates(variables, attributes, non_dim_coord_names):
         # we get support for attrs["coordinates"] for free.
         coords_str = pop_to(encoding, attrs, "coordinates")
         if not coords_str and variable_coordinates[name]:
-            attrs["coordinates"] = " ".join(
+            coordinates_text = " ".join(
                 str(coord_name)
                 for coord_name in variable_coordinates[name]
                 if coord_name not in not_technically_coordinates
             )
+            if len(coordinates_text) > 0:
```

Comment:
`bool(str)` is `True` for any string other than `""`, so we could use
```suggestion
            if coordinates_text != "":
```
or even
```suggestion
            if coordinates_text:
```
comparing the length should be fine, too, though.

Reply:
Second option looks very natural to me. Committed.